app/views/games/show.html.erb

```diff
@@ -66,20 +58,33 @@
         var piece = $.parseHTML(
           '<div class="piece" data-id='+ piece.id +'>' + pieceHtml + '</div>');
 
-        $chessboard.find('.square#'+squareId).append(piece);
-        $chessboard.find('.piece').draggable();
+        $('.square#'+squareId).append(piece);
+        $('.piece').draggable();
       }
+    }
 
-      function movePiece(piece) {
-        var coord = piece.closest('.square').attr('id');
-        var newRow = coord[0];
-        var newCol = coord[1];
-
-        $.post(piecesURL + "/" + piece.data("id"), {
-          _method: "PUT",
-          piece: { row: newRow, col: newCol }
-        }).success();
+    $(".square").droppable({
+      drop: function(event, ui) {
+        var coord = $(this).attr('id').split('');
+        var $piece = $(ui.draggable);
+        movePiece($piece, coord);
       }
     });
+
+    // race condition without setTimeout():
+    // The get request may complete before the post request's back-end logic is
+    //   completely executed.
+    // In the worst case, the get request will retrieve pieces before they are
+    //   updated by Piece#move_to!.
```

Comment:
How is this a race condition?  It should be impossible for the Rails app to send a response before its backend logic has fully completed.  You should be able to replace the success callback with simply:

```javascript
.success(function() { setupBoard() });
```

However, one idea would be to return the board directly from the update action in the same JSON format as the pieces URL, then you don't need to execute a second GET request (so an update is a simple PUT with a response showing the new state of the board).  Then you can extract out the board update function from the success callback and use it in both calls:

```javascript

function updateBoard(pieces) {
  pieces.forEach(function(piece, i){
    // ...
  }
}

$.get(piecesURL).success(updateBoard);

$.post(piecesURL + "/" + piece.data("id"), { ... }).success(updateBoard);
```